### PR TITLE
Add ability to include license files in extensions

### PR DESCRIPTION
**Pull request by Vishwas-Adiga**
This PR gives extension developers the ability to include a license file in their work. The feature will work similar to the `iconName` property of the `DesignerComponent` annotation.

Extension developers will be able to add a license in two ways:
1. Add a license file to aiwebres and include it in the `DesignerComponent` annotation
```
@DesignerComponent(version = 1,
    description = "My Extension",
    category = ComponentCategory.EXTENSION,
    nonVisible = true,
    licenseName = "aiwebres/LICENSE")
```

2. Link to the license that is hosted externally

```
@DesignerComponent(version = 1,
    description = "My Extension",
    category = ComponentCategory.EXTENSION,
    nonVisible = true,
    licenseName = "https://my.website.com/my-license.txt")
```

---
Users will be able to access the license from the palette help widget.
image

Clicking the "View license" link will open it in a new tab

---

Extensions that do not include a license will not show a link, as expected
image

### CODE REVIEW

**Comment by alanscot** [comment]
And how would one use this "License" ?

**Comment by ewpatton** [comment]
@alanscot Some open source licenses require attribution when used in downstream products. Extension authors who make use of FOSS libraries may need to credit the original authors of those libraries. Those terms may extend to app inventors using extensions. It may also be the case that some extension authors want to release their extension under specific licensing terms.

**Comment by jisqyv** [comment]
@Vishwas-Adiga I unassigned you from the PR. The assigned person is who will shepherd and/or review the PR. That cannot be the author.

**Comment by vknow360** [comment]
Amazing :+1:

File: appinventor/appengine/src/com/google/appinventor/server/DownloadServlet.java [hunk 1]
```diff
@@ -244,7 +244,9 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOExc
 
     // Set http response information
     resp.setStatus(HttpServletResponse.SC_OK);
-    resp.setHeader("content-disposition", "attachment; filename=\"" + fileName + "\"");
+    resp.setHeader(
+        "content-disposition",
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/server/DownloadServlet.java:248]: There is extra trailing whitespace on this line.

**Review by ewpatton — CHANGES_REQUESTED** [review]
One minor whitespace issue but otherwise LGTM.

File: appinventor/appengine/src/com/google/appinventor/server/DownloadServlet.java (hunk 1 above)
Comment (Vishwas-Adiga) [reply appinventor/appengine/src/com/google/appinventor/server/DownloadServlet.java]: Fixed 👍

**Review by ewpatton — APPROVED** [review]

**Pull request merged**